packages/metadata/src/providers/cachingProxy.provider.ts

```diff
@@ -3,15 +3,15 @@ import { z } from "zod";
 import { ICache } from "@grants-stack-indexer/repository";
 import { ILogger } from "@grants-stack-indexer/shared";
 
-import { IMetadataProvider } from "../internal.js";
+import { ICacheableMetadataProvider, IMetadataProvider } from "../internal.js";
 
 /**
  * A metadata provider that caches metadata lookups from the underlying provider.
  * When a metadata is requested, it first checks the cache. If found, returns the cached metadata.
  * If not found in cache, fetches from the underlying provider and caches the result before returning.
  * Cache failures (both reads and writes) are logged but do not prevent the provider from functioning.
  */
-export class CachingMetadataProvider implements IMetadataProvider {
+export class CachingMetadataProvider implements ICacheableMetadataProvider {
```

Comment:
excelent place to apply interface segregation principle  :D 

```suggestion
export class CachingMetadataProvider implements ICacheableMetadataProvider, IMetadataProvider{
```